packages/perseus/src/widgets/radio/choice-none-above.jsx

```diff
@@ -1,54 +1,60 @@
 // @flow
 
 import * as i18n from "@khanacademy/wonder-blocks-i18n";
-import classNames from "classnames";
-import PropTypes from "prop-types";
 import * as React from "react";
 
 import Renderer from "../../renderer.jsx";
 
 import Choice from "./choice.jsx";
 
-class ChoiceNoneAbove extends React.Component<$FlowFixMe> {
-    _choice: $FlowFixMe;
-    static propTypes = {
-        className: PropTypes.string,
-        content: PropTypes.node,
-        showContent: PropTypes.bool,
+import type {ChoiceProps} from "./choice.jsx";
+
+type Props = {|
+    ...ChoiceProps,
+    showContent: boolean,
+|};
+
+type WithForwardRef = {|forwardedRef: React.Ref<"button">|};
+
+type PropsWithForwardRef = {|
+    ...Props,
+    ...WithForwardRef,
+|};
+
+function ChoiceNoneAbove(props: PropsWithForwardRef): React.Node {
+    const {showContent, content, forwardedRef, ...rest} = props;
+
+    const choiceProps = {
+        ...rest,
+        content: showContent ? (
+            content
+        ) : (
+            // We use a Renderer here because that is how
+            // `this.props.content` is wrapped otherwise.
+            // We pass in a key here so that we avoid a semi-spurious
+            // react warning when we render this in the same place
+            // as the previous choice content renderer.
+            // Note this destroys state, but since all we're doing
+            // is outputting "None of the above", that is okay.
+            <Renderer
+                key="noneOfTheAboveRenderer"
+                content={i18n._("None of the above")}
+            />
+        ),
     };
 
-    static defaultProps: $FlowFixMe = {
-        showContent: true,
-    };
+    return <Choice {...choiceProps} ref={forwardedRef} />;
+}
 
-    focusInput: () => void = () => {
```

Comment:
I went ahead and refactored ChoiceNoneAbove at the same time as Choice since they're so closely related. Honestly they could probably be merged, but I didn't think about it until just now so I'll follow up later.

I couldn't find any uses of focusInput, so I pulled it out of these components.

Reply:
There's a lot of stuff going on in this PR.  Merging can always happen in a future PR.

Reply:
Since they are so closely related, I also think merging in the future would make sense! Although, would that affect places where one is used without the other? Or are they always used together?